src/driver-roccat-kone-emp.c

```diff
@@ -0,0 +1,1132 @@
+		if(button->action.type == RATBAG_BUTTON_ACTION_TYPE_MACRO) {
+			macro = &drv_data->macros[profile->index][button->index];
+			memset(macro, 0, sizeof(struct roccat_macro));
+
+			macro->reportID = ROCCAT_REPORT_ID_MACRO;
+			macro->bank = ROCCAT_BANK_ID_1;
+			macro->profile = profile->index;
+			macro->button_index = button->index;
+			macro->repeats = 0; // No repeats in libratbag
+
+			if(button->action.macro->group) {
+				// Seems no use of group in libratbag
+				strncpy(macro->group, button->action.macro->group, ROCCAT_MACRO_GROUP_NAME_LENGTH); 
+			} else {
+				strncpy(macro->group, "libratbag macros", ROCCAT_MACRO_GROUP_NAME_LENGTH); 
+			}
+			strncpy(macro->name, button->action.macro->name, ROCCAT_MACRO_NAME_LENGTH); 
+
+			for (i = 0; i < MAX_MACRO_EVENTS && count < ROCCAT_MAX_MACRO_LENGTH; i++) {
+				if (button->action.macro->events[i].type == RATBAG_MACRO_EVENT_INVALID)
+					return -EINVAL; /* should not happen, ever */
+
+				if (button->action.macro->events[i].type == RATBAG_MACRO_EVENT_NONE)
+					break;
+
+				/* ignore the first wait */
+				if (button->action.macro->events[i].type == RATBAG_MACRO_EVENT_WAIT &&
+					!count)
+					continue;
+
+				if (button->action.macro->events[i].type == RATBAG_MACRO_EVENT_KEY_PRESSED ||
+					button->action.macro->events[i].type == RATBAG_MACRO_EVENT_KEY_RELEASED) {
+					macro->keys[count].keycode = ratbag_hidraw_get_keyboard_usage_from_keycode(device, button->action.macro->events[i].event.key);
+				}
+
+				switch (button->action.macro->events[i].type) {
+				case RATBAG_MACRO_EVENT_KEY_PRESSED:
+					macro->keys[count].flag = 0x01;
+					break;
+				case RATBAG_MACRO_EVENT_KEY_RELEASED:
+					macro->keys[count].flag = 0x02;
+					break;
+				case RATBAG_MACRO_EVENT_WAIT:
+					macro->keys[--count].time = button->action.macro->events[i].event.timeout;
+					break;
+				case RATBAG_MACRO_EVENT_INVALID:
+				case RATBAG_MACRO_EVENT_NONE:
+					/* should not happen */
+					log_error(device->ratbag,
+						"something went wrong while writing a macro.\n");
+				}
+				count++;
+			}
+			macro->length = count;
+			
+
+			// Macro has to be send in two packets
+			memcpy(bankBuf, macro, ROCCAT_REPORT_SIZE_MACRO_BANK);
+
+			rc = ratbag_hidraw_set_feature_report(device, ROCCAT_REPORT_ID_MACRO,
+						  bankBuf, ROCCAT_REPORT_SIZE_MACRO_BANK);    
+			if (rc < 0)
+				return rc;
+
+			if (rc != ROCCAT_REPORT_SIZE_MACRO_BANK)
+				return -EIO;
+
+			rc = roccat_wait_ready(device);
+			if (rc)
+				log_error(device->ratbag,
+					"Error while waiting for the device to be ready: %s (%d)\n",
+					strerror(-rc), rc);
+
+			bankBuf[0] = ROCCAT_REPORT_ID_MACRO;
+			bankBuf[1] = ROCCAT_BANK_ID_2;
+			// The remaining macro structure is not big enough to fill the second bank
+			// Write the remaining, fill the end with 0 
+			unsigned int remainingToWrite = sizeof(struct roccat_macro)-ROCCAT_REPORT_SIZE_MACRO_BANK;
```

Comment:
Snake case.

Reply:
Can you be more precise, please. Is there something I need to fix or improve?

Reply:
[Snake case](https://en.wikipedia.org/wiki/Snake_case) for `remainingToWrite`, i. e. `remaining_to_write`.

Reply:
Oof, my bad, I was not thinking at all this way. I was thinking about ouroboros/overwritting data with the buffer and memcpy